### PR TITLE
feature(backend): adding the open search test connection logic

### PR DESCRIPTION
**Pull request by xoscar**
This PR adds the test connection logic for the open search db file

## Changes

- Updates the open search file to test the connection
- Fixes the OpenAPI types and the FE

## Fixes

- #1693 

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

https://www.loom.com/share/e7e3ef7a5957479883f36a123da661d1

### CODE REVIEW

File: api/openapi.yaml [hunk 1]
```diff
@@ -1107,4 +1107,4 @@ paths:
           content:
             application/json:
               schema:
-                $ref: "./config.yaml#/components/schemas/TestConnectionResponse"
+                $ref: "./config.yaml#/components/schemas/ConnectionResult"
```
Comment (xoscar) [inline api/openapi.yaml:1110]: updating this to match the actual response from the service

File: server/tracedb/opensearchdb.go [hunk 2]
```diff
@@ -29,8 +30,52 @@ func (db opensearchDb) Close() error {
...
+		}
+	}
+
+	addressesString := strings.Join(db.config.Addresses, ",")
+	_, err := db.GetTraceByID(ctx, trace.TraceID{}.String())
+
+	if strings.Contains(strings.ToLower(err.Error()), "unauthorized") {
+		return ConnectionTestResult{
+			AuthenticationTestResult: ConnectionTestStepResult{
+				OperationDescription: `Tracetest tried to execue an OpenSearch API request but it failed due to authentication issues`,
```
Comment (mathnogueira) [inline server/tracedb/opensearchdb.go:53]:
```suggestion
				OperationDescription: `Tracetest tried to execute an OpenSearch API request but it failed due to authentication issues`,
```

**Review by mathnogueira — APPROVED** [review]
Looks good

**Review by jorgeepc — APPROVED** [review]
Looks good!

**Pull request merged**